File: src/main/java/gg/mineral/practice/commands/config/KitEditorCommand.java
package gg.mineral.practice.commands.config;

import gg.mineral.practice.commands.PlayerCommand;

import gg.mineral.practice.PracticePlugin;
import gg.mineral.practice.kit.KitEditorManager;
import gg.mineral.practice.managers.GametypeManager;
import gg.mineral.practice.managers.PlayerManager;
import gg.mineral.practice.managers.QueuetypeManager;
import gg.mineral.practice.util.messages.ChatMessages;
import gg.mineral.practice.util.messages.ErrorMessages;
import gg.mineral.practice.util.messages.UsageMessages;

public class KitEditorCommand extends PlayerCommand {
	final KitEditorManager kitEditorManager = PracticePlugin.INSTANCE.getKitEditorManager();;
	final PlayerManager playerManager = PracticePlugin.INSTANCE.getPlayerManager();
	final GametypeManager gametypeManager = PracticePlugin.INSTANCE.getGametypeManager();
	final QueuetypeManager queuetypeManager = PracticePlugin.INSTANCE.getQueuetypeManager();

	public KitEditorCommand() {
		super("kiteditor", "practice.config");
	}

	@Override
	public void execute(org.bukkit.entity.Player player, String[] args) {

		String arg = args.length > 0 ? args[0] : "";

		switch (arg.toLowerCase()) {
			default:
				ChatMessages.KIT_EDITOR_COMMANDS.send(player);
				ChatMessages.KIT_EDITOR_ENABLE.send(player);
				ChatMessages.KIT_EDITOR_DISPLAY.send(player);
				ChatMessages.KIT_EDITOR_SLOT.send(player);
				ChatMessages.KIT_EDITOR_LOCATION.send(player);

				return;
			case "enable":
				if (args.length < 2) {
					UsageMessages.KIT_EDITOR_ENABLE.send(player);
					return;
				}

				String toggled = args[1].toLowerCase();

				switch (toggled) {
					case "false":
						kitEditorManager.setEnabled(false);
						break;
					case "true":
						kitEditorManager.setEnabled(true);
						break;
					default:
						UsageMessages.KIT_EDITOR_ENABLE.send(player);
						return;
				}

				ChatMessages.KIT_EDITOR_ENABLED.clone()
						.replace("%toggled%", toggled).send(player);

				return;
			case "setdisplay":
				if (args.length < 1) {
					UsageMessages.KIT_EDITOR_DISPLAY.send(player);
					return;
				}

				kitEditorManager.setDisplayItem(player.getItemInHand());

				if (args.length > 1) {
					String name = args[1].replace("&", "§");
					kitEditorManager.setDisplayName(name);
				}

				ChatMessages.KIT_EDITOR_DISPLAY_SET.send(player);

				return;
			case "slot":
				if (args.length < 2) {
					UsageMessages.KIT_EDITOR_SLOT.send(player);
					return;
				}

				int slot;
				String strSlot = args[1];
				try {
					slot = Integer.parseInt(strSlot);
				} catch (Exception e) {
					ErrorMessages.INVALID_SLOT.send(player);
					return;
				}

				kitEditorManager.setSlot(slot);

				ChatMessages.KIT_EDITOR_SLOT_SET.clone().replace("%slot%", strSlot).send(player);
				;

				return;
			case "setlocation":
				kitEditorManager.setLocation(player.getLocation());
				ChatMessages.KIT_EDITOR_LOCATION_SET.send(player);

				return;
		}
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/config/KitEditorCommand.kt
package gg.mineral.practice.commands.config

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import dev.rollczi.litecommands.annotations.permission.Permission
import gg.mineral.practice.entity.appender.CommandSenderAppender
import gg.mineral.practice.managers.KitEditorManager
import gg.mineral.practice.managers.KitEditorManager.displayItem
import gg.mineral.practice.managers.KitEditorManager.enabled
import gg.mineral.practice.managers.KitEditorManager.location
import gg.mineral.practice.util.messages.impl.ChatMessages
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*

@Command(name = "kiteditor")
@Permission("practice.config")
class KitEditorCommand : CommandSenderAppender {

    @Execute(name = "enable")
    fun executeEnable(@Context sender: CommandSender, @Arg toggled: Boolean) {
        enabled = toggled
        sender.send(
            ChatMessages.KIT_EDITOR_ENABLED.clone()
                .replace("%toggled%", toggled.toString())
        )
    }

    @Execute(name = "setdisplay")
    fun executeSetDisplay(@Context player: Player, @Arg displayName: Optional<String>) {
        displayName.ifPresent { KitEditorManager.displayName = it.replace("&", "§") }

        displayItem = player.itemInHand

        player.send(ChatMessages.KIT_EDITOR_DISPLAY_SET)
    }

    @Execute(name = "slot")
    fun executeSlot(@Context sender: CommandSender, @Arg slot: Int) {
        KitEditorManager.slot = slot
        sender.send(ChatMessages.KIT_EDITOR_SLOT_SET.clone().replace("%slot%", slot.toString()))
    }

    @Execute(name = "setlocation", aliases = ["location"])
    fun executeLocation(@Context player: Player) {
        location = player.location
        player.send(ChatMessages.KIT_EDITOR_LOCATION_SET)
    }

    @Execute(aliases = ["help"])
    fun executeHelp(@Context sender: CommandSender) {
        sender.send(
            ChatMessages.KIT_EDITOR_COMMANDS,
            ChatMessages.KIT_EDITOR_ENABLE,
            ChatMessages.KIT_EDITOR_DISPLAY,
            ChatMessages.KIT_EDITOR_SLOT,
            ChatMessages.KIT_EDITOR_LOCATION
        )
    }
}
